src/main/java/org/candlepin/resource/DeletedConsumerResource.java

```diff
@@ -33,21 +40,56 @@
  */
 public class DeletedConsumerResource implements DeletedConsumerApi {
     private final DeletedConsumerCurator deletedConsumerCurator;
+    private final I18n i18n;
     private final ModelTranslator translator;
+    private final Configuration config;
 
     @Inject
-    public DeletedConsumerResource(DeletedConsumerCurator deletedConsumerCurator,
-        ModelTranslator translator) {
+    public DeletedConsumerResource(DeletedConsumerCurator deletedConsumerCurator, I18n i18n,
+        ModelTranslator translator, Configuration config) {
+        this.i18n = Objects.requireNonNull(i18n);
         this.deletedConsumerCurator = Objects.requireNonNull(deletedConsumerCurator);
         this.translator = Objects.requireNonNull(translator);
+        this.config = Objects.requireNonNull(config);
     }
 
     @Override
-    public Stream<DeletedConsumerDTO> listByDate(OffsetDateTime date) {
-        List<DeletedConsumer> deletedConsumers = date != null ?
-            this.deletedConsumerCurator.findByDate(Util.toDate(date)) :
-            this.deletedConsumerCurator.listAll();
-        return deletedConsumers.stream()
+    public Stream<DeletedConsumerDTO> listByDate(OffsetDateTime date, Integer page, Integer perPage,
+        String order, String sortBy) {
+        PageRequest pageRequest = ResteasyContext.getContextData(PageRequest.class);
+        DeletedConsumerQueryArguments queryArgs = new DeletedConsumerQueryArguments();
+        long count = this.deletedConsumerCurator.getDeletedConsumerCount(queryArgs);
+
+        if (pageRequest != null) {
+            Page<Stream<DeletedConsumerDTO>> pageResponse = new Page<>();
+            pageResponse.setPageRequest(pageRequest);
+
+            if (pageRequest.isPaging()) {
+                queryArgs.setOffset((pageRequest.getPage() - 1) * pageRequest.getPerPage())
+                    .setLimit(pageRequest.getPerPage());
+            }
+
+            if (pageRequest.getSortBy() != null) {
+                boolean reverse = pageRequest.getOrder() == PageRequest.Order.DESCENDING;
```

Comment:
Instead of explicitly setting the order as descending, lets use the default variable (in case we ever change the default): PageRequest.DEFAULT_ORDER